Increase test coverage

jvdassen opened the pull request:
Hello everyone,

We have increase the scope of the unit testing in this project with focus on the `classify`, `math`, `process`, `simple` and `util` packages.
As you can see there were no modifications to the `src` of the project and we did not incorporate new dependencies as requested in [PR 1022](https://github.com/stanfordnlp/CoreNLP/issues/1022). 
All tests were written using the existing JUnit version and aligned to the formatting of the existing classes.

As requested in the [contribution guidelines](https://github.com/stanfordnlp/CoreNLP/blob/master/CONTRIBUTING.md) we are happy to declare this contribution under public domain.

AngledLuffa commented:
I'm going to start going through this, but I'd appreciate it if you could squash some of the changes down and set your whitespace to use 2 spaces instead of tabs.

AngledLuffa commented on test/src/edu/stanford/nlp/simple/TokenTest.java:
```diff
@@ -79,13 +78,19 @@ public void testCharacterOffsetEnd() {
 
     @Test
-    public void testWhiteSpace() {
+    public void testWhiteSpaceBefore() {
```
This really doesn't seem necessary

AngledLuffa commented on test/src/edu/stanford/nlp/dcoref/RulesTest.java:
```diff
@@ -20,12 +19,12 @@
-    assertFalse(Rules.isAcronym(IBM2, IBMM));
-    assertFalse(Rules.isAcronym(IBM, MIBM));
-    assertFalse(Rules.isAcronym(IBM2, MIBM));
+    assertTrue("Acronym IMB -> International Business Machines",Rules.isAcronym(IBM, IBM2));
```
You get a line number for where the test fails so I don't really agree this is an mystery assertion.  If you want to gather up a bunch of these changes into a single change, though, I can merge it

jvdassen commented:
Thanks for the feedback! Unfortunately well have to create a new PR to change the upstream branch to the one where we fixed and squashed our contributions!
